sql/src/main/java/org/elasticsearch/action/bulk/SymbolBasedBulkShardProcessor.java

```diff
@@ -273,6 +302,64 @@ private void executeRequests() {
         }
     }
 
+    private void createPendingIndices() {
+        final List<PendingRequest> pendings = new ArrayList<>();
+        final Set<String> indices;
+
+        synchronized (requestsForNewIndices) {
+            indices = ImmutableSet.copyOf(Sets.difference(requestsForNewIndices.keySet(), indicesCreated));
+            for (Map.Entry<String, List<PendingRequest>> entry : requestsForNewIndices.entrySet()) {
+                pendings.addAll(entry.getValue());
+            }
+            requestsForNewIndices.clear();
+            pendingNewIndexRequests.set(0);
+        }
+
+        if (indices.size() > 0 && Iterables.any(indices, shouldAutocreateIndexPredicate)) {
+            LOGGER.debug("create {} pending indices in bulk...", indices.size());
+
+            final String[] indicesNames = indices.toArray(new String[indices.size()]);
+
+            BulkCreateIndicesRequest bulkCreateIndicesRequest = new BulkCreateIndicesRequest(indicesNames)
```

Comment:
please change the of BulkCreateIndicesRequest to take a Collection... there are unnecessary array/collection transformations otherwise as you don't really use a string array inside anyway